namerd/iface/interpreter-thrift/src/main/scala/io/buoyant/namerd/iface/ThriftNamerClient.scala

```diff
@@ -3,19 +3,21 @@ package io.buoyant.namerd.iface
 import com.twitter.finagle.Name.Bound
 import com.twitter.finagle._
 import com.twitter.finagle.naming.NameInterpreter
+import com.twitter.finagle.stats.DefaultStatsReceiver
 import com.twitter.finagle.tracing.Trace
 import com.twitter.finagle.util.DefaultTimer
 import com.twitter.io.Buf
 import com.twitter.logging.Logger
 import com.twitter.util._
 import com.twitter.util.TimeConversions._
-import io.buoyant.namer.{Metadata, DelegateTree, Delegator}
+import io.buoyant.namer.{DelegateTree, Delegator, Metadata}
 import io.buoyant.namerd.iface.{thriftscala => thrift}
 import java.net.{InetAddress, InetSocketAddress}
 
 class ThriftNamerClient(
   client: thrift.Namer.FutureIface,
   namespace: String,
+  statsPrefix: String = "",
```

Comment:
I think what you want is to pass a statsReceiver into the constructor here.  In NamerdInterpreterInitializer's `newInterpreter` you can extract the stats receiver from the params and pass it in.  At that point you can also scope it to indicate that it's for the namerd interpreter.